packages/client/src/utils.ts

```diff
@@ -73,3 +76,49 @@ export function encodeCsvRow(items: any[]): string {
     })
     .join(",");
 }
+
+/**
+ * Given a observation series `num`, computes the ratio of the each
+ * observation value in `num` to the observation value in `denom` with
+ * the closest date value.
+ *
+ * Both `num` and `denom` series are sorted.
+ *
+ * @param num Sorted numerator observation series.
+ * @param denom Sorted denominator time series.
+ *
+ * @returns A list of Observations with the per capita calculation applied to its values.
+ */
+export function computeRatio(
+  num: Observation[],
+  denom: Observation[],
+  scaling = 1
+): Observation[] {
+  if (_.isEmpty(denom)) {
+    return [];
+  }
+  const result: Observation[] = [];
+  let j = 0; // denominator position
+  for (let i = 0; i < num.length; i++) {
+    const numDate = Date.parse(num[i].date);
+    const denomDate = Date.parse(denom[j].date);
+    while (j < denom.length - 1 && numDate > denomDate) {
```

Comment:
nit: add a quick comment on what the while loop does, so it's easier for future maintainers to parse this function at a glance. e.g. `// find denom with closest date to num`

Reply:
Done!